openbb_sdk/providers/fmp/openbb_fmp/models/market_snapshots.py

```diff
@@ -0,0 +1,160 @@
+    "SES",
+    "SET",
+    "SGO",
+    "SHH",
+    "SHZ",
+    "SIX",
+    "STO",
+    "TAI",
+    "TLV",
+    "TSX",
+    "TWO",
+    "VIE",
+    "WSE",
+    "XETRA",
+]
+
+
+class FMPMarketSnapshotsQueryParams(MarketSnapshotsQueryParams):
+    """FMP Market Snapshots Query.
+
+    Source: https://site.financialmodelingprep.com/developer/docs/#Most-of-the-EuroNext
+    """
+
+    market: MARKETS = Field(
+        description="The market to fetch data for.", default="NASDAQ"
+    )
+
+
+class FMPMarketSnapshotsData(MarketSnapshotsData):
+    """FMP Market Snapshots Data."""
+
+    class Config:
+        fields = {
+            "high": "dayHigh",
+            "low": "dayLow",
+            "prev_close": "previousClose",
+            "change_percent": "changesPercentage",
+        }
+
+    avg_volume: Optional[int] = Field(
+        description="Average volume of the stock.", alias="avgVolume"
+    )
+    ma50: Optional[float] = Field(
+        description="The 50-day moving average.", alias="priceAvg50"
+    )
+    ma200: Optional[float] = Field(
+        description="The 200-day moving average.", alias="priceAvg200"
+    )
+    year_high: Optional[float] = Field(
+        description="The 52-week high.", alias="yearHigh"
+    )
+    year_low: Optional[float] = Field(description="The 52-week low.", alias="yearLow")
+    market_cap: Optional[float] = Field(
+        description="Market cap of the stock.", alias="marketCap"
+    )
+    shares_outstanding: Optional[int] = Field(
+        description="Number of shares outstanding.", alias="sharesOutstanding"
+    )
+    eps: Optional[float] = Field(description="Earnings per share.")
+    pe: Optional[float] = Field(description="Price to earnings ratio.")
+    exchange: Optional[str] = Field(description="The exchange of the stock.")
+
+
+class FMPMarketSnapshotsFetcher(
+    Fetcher[
+        FMPMarketSnapshotsQueryParams,
+        List[FMPMarketSnapshotsData],
+    ]
+):
+    """Transform the query, extract and transform the data from the FMP endpoints."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> FMPMarketSnapshotsQueryParams:
+        """Transform the query params."""
+        return FMPMarketSnapshotsQueryParams(**params)
+
+    @staticmethod
+    def extract_data(
+        query: FMPMarketSnapshotsQueryParams,
+        credentials: Optional[Dict[str, str]],
+        **kwargs: Any,
+    ) -> List[Dict]:
+        """Return the raw data from the FMP endpoint."""
+
+        api_key = credentials.get("fmp_api_key") if credentials else ""
+        url = f"https://financialmodelingprep.com/api/v3/quotes/{query.market}?apikey={api_key}"
+        response = make_request(url)
+
+        if response.status_code != 200:
+            raise RuntimeError(
+                f"Error fetching data from FMP  -> {response.status_code}"
+            )
+        data = pd.DataFrame(response.json())
+        data = data.drop(columns=["timestamp", "earningsAnnouncement"]).fillna(value=0)
```

Comment:
These items were causing headaches because they are not uniform across all markets and can create errors.  Fillna is also included to be compatible with Rest.

Reply:
@IgorWounds thoughts?

Reply:
Other alternative is to encode those fields as strings, but that would go against other policies.

Reply:
Right. It would be a messy hack to debug if problems arise.

Reply:
The general rule is that our standardization framework shouldn't do any changes to the data in terms of dropping things, calculating things, or filling in missing things. We just return the data and do any transformations in terms of types that are needed. This way, the provider is the one that is responsible for the data consistency and quality and not OpenBB which just returns that data.

Reply:
Yes, we can make them strings. Am I correct that they are not a part of our standard?

Reply:
You are correct:

```python

class MarketSnapshotsData(Data):
    """Market Snapshots Data"""

    symbol: str = Field(description="The stock symbol.")

    open: Optional[float] = Field(
        description="The opening price of the stock on the current trading day."
    )
    high: Optional[float] = Field(
        description="The highest price of the stock on the current trading day."
    )
    low: Optional[float] = Field(
        description="The lowest price of the stock on the current trading day."
    )
    prev_close: Optional[float] = Field(
        description="The previous closing price of the stock on the current trading day."
    )
    change: Optional[float] = Field(
        description="The change in price over the current trading day."
    )
    change_percent: Optional[float] = Field(
        description="The change, as a percent, over the current trading day."
    )
    volume: Optional[int] = Field(
        description="The volume of the stock on the current trading day."
    )
```

**note**: The models will need to be updated for the switch to Pydantic V2, and then there's the directory rename thing too.  This one is a bit behind.